File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceContainerLowLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerLowLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl11
            AppTheme.STANDARD -> R.color.cstl11
            AppTheme.BLUE -> R.color.cbl11
            AppTheme.GREEN -> R.color.cgl11
            AppTheme.RED -> R.color.crl11
            AppTheme.NOTHING -> R.color.cnol11
        }
    }
}

File: lib/layer2_core/light/e_surface_container_low_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceContainerLowLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl11;
      case 'standard': return cstl11;
      case 'blue': return cbl11;
      case 'green': return cgl11;
      case 'red': return crl11;
      case 'nothing': return cnol11;
      default: return cstl11;
    }
  }
}
